Source/Charts/Utils/ChartUtils.swift

```diff
@@ -41,50 +75,7 @@ open class ChartUtils
         internal static let DEG2RAD = Double.pi / 180.0
         internal static let RAD2DEG = 180.0 / Double.pi
     }
-    
-    internal class func roundToNextSignificant(number: Double) -> Double
-    {
-        if number.isInfinite || number.isNaN || number == 0
-        {
-            return number
-        }
-        
-        let d = ceil(log10(number < 0.0 ? -number : number))
-        let pw = 1 - Int(d)
-        let magnitude = pow(Double(10.0), Double(pw))
-        let shifted = round(number * magnitude)
-        return shifted / magnitude
-    }
-    
-    internal class func decimals(_ number: Double) -> Int
-    {
-        if number.isNaN || number.isInfinite || number == 0.0
-        {
-            return 0
-        }
-        
-        let i = roundToNextSignificant(number: Double(number))
-        
-        if i.isInfinite || i.isNaN
-        {
-            return 0
-        }
-        
-        return Int(ceil(-log10(i))) + 2
-    }
-    
-    internal class func nextUp(_ number: Double) -> Double
```

Comment:
swift doc:
>For any finite value x, x.nextUp is greater than x. For nan or infinity, x.nextUp is x itself. The following special cases also apply:
>
>If x is -infinity, then x.nextUp is -greatestFiniteMagnitude.
If x is -leastNonzeroMagnitude, then x.nextUp is -0.0.
If x is zero, then x.nextUp is leastNonzeroMagnitude.
If x is greatestFiniteMagnitude, then x.nextUp is infinity.

I'm kind of confused `If x is -infinity, then x.nextUp is -greatestFiniteMagnitude.` because `For nan or infinity, x.nextUp is x itself`

Reply:
The only place this is used is in the axis renderers. I removed this implementation because the default one is safer to pass into the call sites.

Reply:
yep but it's different than the library implementation, towards the infinity check and result. I need to check several places.

Reply:
but the real issue is swift doc has paradox:
>For nan or infinity, x.nextUp is x itself. The following special cases also apply

does infinity here contains both +inf and -inf? or just +inf?

because it added at last line:
>If x is -infinity, then x.nextUp is -greatestFiniteMagnitude

Reply:
`.infinitiy != -.infinity`
It makes sense that the nextUp from negative infinity is the negative greatest finite magnitude. The opposite is true for `nextDown()`